librad/src/net/protocol.rs

```diff
@@ -66,6 +66,7 @@ mod tick;
 pub struct Config {
     pub paths: Paths,
     pub listen_addr: SocketAddr,
+    pub advertised_addrs: Option<Vec<SocketAddr>>,
```

Comment:
Sorry for nitpicking, but `Option<Vec<T>>` is not a terribly useful type: if it’s `None`, we know the list has no elements, but if it’s `Some` it may also not have any elements.

Since the list is allowed to be empty, `Vec` would do, otherwise we usually use `NonEmpty`

Reply:
That's a very good point. I want to keep option so that I can use .map and others. I'll make it NonEmpty instead.